src/ex1/script.js

```diff
@@ -0,0 +1,81 @@
+const newToDo_txt = document.getElementById("NewToDo_txt");
+const newToDo_btn = document.getElementById("NewToDo_btn");
+const tasksList = document.getElementById("taskList");
+const clearAll_btn = document.getElementById("clearAll");
+const emptyState = document.getElementById("emptyState");
+const amountTasks = document.getElementById("amountTasks");
+let taskCount = 0;
+
+newToDo_btn.addEventListener("click", () => {
+	if (newToDo_txt.value) {
+		const taskElement = document.createElement("div");
+		taskElement.innerText = newToDo_txt.value;
+		taskElement.className = "taskElement clickable";
+		taskElement.addEventListener("click", () => {
```

Comment:
better create a dedicated function for each operation such as: `onTaskClick` instead of creating an arrow function every time you create an element (which will create a different anonymous function in memory each time)

Reply:
אני אשמח אם יש הסבר הגיוני למה לעשות את זה בבקשה כי זה מרגיש לי עודף עבודה מיותר גם לי וגם למחשב ולכן אני לא משתמש בזה אם תוכל בבקשה להסביר לי את ההיגיון העומד מאחורי זה אשמח

Reply:
1. מבחינת זיכרון -  אם יצרת 100 אלמנטים גם יצרת 100 פונקציות בזיכרון בצורה הזאת. (ולכן דווקא למחשב שלך זה כן עודף עבודה וגם למחשב של מי שישתמש באתר שלך.  כמובן אגב שהיום עם הכח מחשוב והכל זה מינורי ממש אבל זה פתח לבאגים וחוסר אופטימיזציה באפליקציות יותר מורכבות בעתיד)
2. לאורך זמן עושים שימוש בפונקציות בעוד מקומות, אם הפונקציה הזאת עושה כמה פעולות ובעתיד את אותן פעולות תרצה להפעיל גם ממקום אחר אז כבר תהיה לך פונקציה קיימת בקוד (בצורה הזאת איך לך פונקציה קיימת)
3. מבחינת best practice עדיף שפונקציות יהיו מתועדות, עם חתימה ושם משלהן (בעתיד כשתעבוד עם typescript או שפות מבוססות type אתה תרצה לדעת מה החתימה (מה הפרמטרים, מה הפונקציה מחזירה וכ׳)